palm/src/ctinfo.c

```diff
@@ -165,14 +170,31 @@ static void PrintLocation(struct ctinfo *info)
 {
     if (info->line != NULL) {
         // We push 4 spaces to take the ' |> ' into account.
-        fprintf(stderr, " |> %s\n    ", info->line);
+        fprintf(stderr, " |> ");
+
+        for (int i = 0; i < info->first_column - 1 && info->line[i] != '\0'; ++i) {
+            fputc(info->line[i], stderr);
+        }
+        if (current_type == CTI_ERROR) fprintf(stderr, RED);
+        else if (current_type == CTI_WARN) fprintf(stderr, LIGHT_PURPLE);
+
+        for (int i = info->first_column - 1; i < info->last_column && info->line[i] != '\0'; ++i) {
+            fputc(info->line[i], stderr);
+        }
+
+        fprintf(stderr, RESET);
+        for (int i = info->last_column; info->line[i] != '\0'; ++i) {
```

Comment:
This code does not work well if last_column is unset (it is optional after all). The reference compiler for example does not set last_column in the parser, and the error message for the following code:
```c
void foo() {
    1;
}
```
is now changed from:
```
error: Error parsing source code: syntax error
 --> test/core/check_error/expr_statement.cvc:2:6
 |>     1;
         ^
```
to:
```
error: Error parsing source code: syntax error
 --> test/core/check_error/expr_statement.cvc:2:6
 |>     1    1;
         ^
```